sbc_compassion/tools/import_letter_functions.py

```diff
@@ -21,13 +21,122 @@
 import checkboxreader as cbr
 import bluecornerfinder as bcf
 import numpy as np
+from math import ceil
 from glob import glob
 from wand.image import Image
+from collections import namedtuple
+
+from openerp import _
 
 
 ##########################################################################
 #                           GENERAL METHODS                              #
 ##########################################################################
+def testline(env, line_vals, csv_file_ids, doc_name):
+    """ Test each line and verify the data given by the csv.
+    If a line does not correspond to the csv, this function erases it.
+    A csv file need to be written in the same way than
+    sbc_compassion/tests/testdata/import/travis_file.csv
+
+    :param env: Odoo environment
+    :param dict line_vals: Dictionnary that will be used to create line
+    :param Recordset('ir.attachment') csv_file_ids: CSV used to check
+    :param str doc_name: Document name
+    :returns: test (True if equal to csv), error
+    :rtype: bool, str
+    """
+    # str that will be returned
+    error = ''
+    csv_data = []
+    # creates a list containing all the csv data
+    for csv_ in csv_file_ids:
+        if os.path.splitext(csv_.name)[1] != '.csv':
+            raise Warning(_("File '{}' is not a csv file".format(csv_.name)))
+        csv_string = base64.b64decode(csv_.datas).splitlines()
+        csv_data += read_csv(csv_string)
+    # check that the letter is present in the csv file and find it
+    values = {}
+    for csv_line in csv_data:
+        if csv_line['name'] == doc_name:
+            values = csv_line
+            break
+    else:
+        raise Warning(
+            _("File '{}' not found in the csv files".format(doc_name)))
+    # Test that the template is the same as given in the csv
+    template_csv = env.ref(
+        'sbc_compassion.'+values['template_id'])
+    tpl_detected = env['sponsorship.correspondence.template'].search(
```

Comment:
I think you don't need to search for the template. You can simply do:
`if template_csv.id != line_vals['template_id']:`